pkg/registry/core/pod/storage/eviction.go

```diff
@@ -201,6 +206,15 @@ func (r *EvictionREST) Create(ctx context.Context, name string, obj runtime.Obje
 
 		pdb := &pdbs[0]
 		pdbName = pdb.Name
+
+		// If the pod is not ready, it doesn't count towards healthy and we should not
+		// decrement.  IsPodHealthy requires a v1Pod, so if we get some other type
+		// of pod, we cannot accurately determine pod health.
+		if validV1Pod && !disruption.IsPodHealthy(v1Pod) && pdb.Status.CurrentHealthy >= pdb.Status.DesiredHealthy && pdb.Status.DesiredHealthy > 0 {
+			updateDeletionOptions = true
```

Comment:
i'm not following this update, if we are about to return do we need to set this variable? (i didn't see a defer'd function)

Reply:
This code is inside an anonymous function: https://github.com/kubernetes/kubernetes/pull/94381/files#diff-9644c646d1076728c605baed12a5813aR189

So, that's the code we're returning from.

Reply:
ahh, thanks! i knew i was missing something